### PR TITLE
expose jenkins plugin e2e as ci-operator step

### PR DESCRIPTION
**Pull request by gabemontero**

### CODE REVIEW

**Comment by gabemontero** [comment]
`* Container test is not ready with reason ImagePullBackOff and message Back-off pulling image "image-registry.openshift-image-registry.svc:5000/ci-op-z1kcyjxy/stable:jenkins-sync-plugin-scripts"`

but the `openshift-image-mirror-mappings` only tweaked the 4.9 branch for some reason TBD (maybe the existing promotion job stuff that was already there)

for the 4.9, the image pull error was "more specific"

`* 2x kubelet: Failed to pull image "image-registry.openshift-image-registry.svc:5000/ci-op-yklrz4v7/stable:jenkins-sync-plugin-scripts": rpc error: code = Unknown desc = Error reading manifest jenkins-sync-plugin-scripts in image-registry.openshift-image-registry.svc:5000/ci-op-yklrz4v7/stable: manifest unknown: manifest unknown`

like the mirroring happened but was quite right or complete

**Comment by gabemontero** [comment]
looks like aws flakes in the 4.10 job

plus we got bit by the inadvertent  pj-rehearse-blocking discussed in announce-testplatform on slack

**Comment by gabemontero** [comment]
a networking hiccup in core jenkins during the 4.10 sync plugin e2e's while executing an agent test caused the failure ... i.e. a "flake" outside our control

`Caused: java.io.IOException: Remote call on JNLP4-connect connection from ip-10-131-0-43.us-east-2.compute.internal/10.131.0.43:41166 failed`

I'll 

/retest

but we can move forward

**Comment by gabemontero** [comment]
/assign @adambkaplan 

I think we are good for "phase 1" as we discussed in office hours 

I'll assign Clayton in a sec

File: core-services/image-mirroring/openshift/mapping_origin_4_9
Comment (gabemontero) [inline core-services/image-mirroring/openshift/mapping_origin_4_9:138]: /assign @smarterclayton 

we are exposing e2e's in the jenkins sync plugin so that they can be consumed by the openshift/jenkins repo (i.e. the one for our openshift jenkins image

doing this based on @adambkaplan experience with step registry and the PR https://github.com/openshift/release/pull/20806 you approved for him last month

long long term ... we envision this as a step in the path to removing the jenkins image from the payload (via decoupling from the openshift/origin suites)

**Comment by gabemontero** [comment]
ok all green CI @adambkaplan

**Comment by gabemontero** [comment]
maven flake on the 4.10 image rehearsal run https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/21474/rehearse-21474-pull-ci-openshift-jenkins-sync-plugin-release-4.10-images/1435309092049522688

**Comment by gabemontero** [comment]
more maven pain

/test pj-rehearse

**Comment by gabemontero** [comment]
sanity check:  the maven download these rehearsals keep failing on is there:  https://repo.jenkins-ci.org/ui/packages/gav:%2F%2Forg.jenkins-ci.plugins:apache-httpcomponents-client-4-api/4.5.13-1.0?name=apache-httpcomponents-client-4-api&type=packages&activeTab=builds

```
[ERROR] Failed to execute goal org.jenkins-ci.tools:maven-hpi-plugin:3.17:resolve-test-dependencies (default-resolve-test-dependencies) on project openshift-sync: Execution default-resolve-test-dependencies of goal org.jenkins-ci.tools:maven-hpi-plugin:3.17:resolve-test-dependencies failed: Failed to resolve org.jenkins-ci.plugins:apache-httpcomponents-client-4-api:hpi:4.5.13-1.0: Could not transfer artifact org.jenkins-ci.plugins:apache-httpcomponents-client-4-api:hpi:4.5.13-1.0 from/to repo.jenkins-ci.org (https://repo.jenkins-ci.org/public/): GET request of: org/jenkins-ci/plugins/apache-httpcomponents-client-4-api/4.5.13-1.0/apache-httpcomponents-client-4-api-4.5.13-1.0.hpi from repo.jenkins-ci.org failed
[ERROR]   org.jenkins-ci.plugins:apache-httpcomponents-client-4-api:hpi:4.5.13-1.0
[ERROR] 
[ERROR] from the specified remote repositories:
[ERROR]   central (https://repo1.maven.org/maven2, releases=true, snapshots=true),
[ERROR]   incremental-repo.jenkins-ci.org (https://repo.jenkins-ci.org/incrementals/, releases=true, snapshots=true),
[ERROR]   repo.jenkins-ci.org (https://repo.jenkins-ci.org/public/, releases=true, snapshots=true),
[ERROR]   nexus-oss-snapshots (https://oss.sonatype.org/content/repositories/snapshots/, releases=false, snapshots=true): Premature end of Content-Length delimited message body (expected: 1,769,549; received: 42,433)
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
STEP 7: FROM image-registry.openshift-image-registry.svc:5000/ci-op-571s8j02/pipeline@sha256:32baa1e0fb419d4c27af630f0c89db9ccd8b7751821bdf654f0700e3ede7e0f1
error: build error: error building at STEP "RUN export PATH=/opt/rh/rh-maven35/root/usr/bin:$PATH && mvn clean package": error while running runtime: exit status 1
```

**Comment by gabemontero** [comment]
ok jenkins artifactory hiccups are gone

bump @adambkaplan @smarterclayton  - can I get this reviewed, let me know if you all want changes, or get tagged for merge

this is for modernizing jenkins CI, where we define the plugin e2e's in the plugin repos themselves, then reference them in the image e2e's .... longer term stepping stone for getting jenkins image out of OCP payload

**Review by adambkaplan — APPROVED** [review]
/lgtm

**Comment by gabemontero** [comment]
bump @smarterclayton 

trying to "modernize" the openshift jenkins CI a bit, with some OCP decoupling elements that are also an eye toward removing the jenkins image from the payload

File: core-services/image-mirroring/openshift/mapping_origin_4_10
Comment (smarterclayton) [inline core-services/image-mirroring/openshift/mapping_origin_4_10:134]: Is this "sync-plugin" for "jerkins"?  Or "Jenkins scripts that implement a sync plugin" for "Jenkins"?  Just want to understand the name so I can piece together a naming scheme.

Comment (gabemontero) [reply core-services/image-mirroring/openshift/mapping_origin_4_10:134]: > Is this "sync-plugin" for "jerkins"? Or "Jenkins scripts that implement a sync plugin" for "Jenkins"? Just want to understand the name so I can piece together a naming scheme.

yes the code we are trying to test is the sync-plugin for jenkins ... the old https://github.com/openshift/jenkins-sync-plugin that dev ex took over from the fabric guys back in the early 3.x days a few years ago.

That jenkins plugin gets included in the set of jenkins plugings included in the openshift jenkins image at https://github.com/openshift/jenkins

I now have the golang based e2e's for it in that actual repo .... PR tests in that repo use those tests.  I'd like for the PR tests in https://github.com/openshift/jenkins to use those tests as well.  This image is suppose to contain the code for those tests in its content.

The "scripts" element comes from my attempt to work off naming conventions @adambkaplan followed with https://github.com/openshift/release/pull/20806 that he got approval from you on @smarterclayton 

Of course if my naming attempts are not ideal, please advise.

OK hopefully that additional context provides what you need @smarterclayton  ... thanks

Comment (adambkaplan) [reply core-services/image-mirroring/openshift/mapping_origin_4_10:134]: @gabemontero why do we need to mirror the scripts image to quay?

My understanding is that the "scripts" image will contain our sync plugin e2e tests which get invoked by `go test` or equivalent. The promotion stanzas instruct CI to make the image available via the `stable` imagestream. We on

Mirroring to quay is a separate thing that is customer facing. This image should only be used in our CI environment - I don't think we need to push it out to end users.

Comment (gabemontero) [reply core-services/image-mirroring/openshift/mapping_origin_4_10:134]: IIRC @adambkaplan (it has been a while since I've gotten attention on this) the change stemmed from running 

`make openshift-image-mirror-mappings `

likewise IIRC I needed to do this after adding https://github.com/openshift/release/pull/21474/files#diff-d039dd1d4471aed28d6c59497dba358abeeb9611c6dd122de1d60acf6d659a37R21 otherwise some of the verification failed

but again, it has been a while

I'll drop that second comment, update the PR, and see what sorts out

if it is not needed, apologies @smarterclayton for the unneeded churn

Comment (gabemontero) [reply core-services/image-mirroring/openshift/mapping_origin_4_10]: ok I have removed the `make openshift-image-mirror-mappings` commit and this file is no longer updated @adambkaplan @smarterclayton

Comment (adambkaplan) [reply core-services/image-mirroring/openshift/mapping_origin_4_10]: Image mirror mapping jobs shouldn't fail here - it seems that the make command logic is trying to create a mirror mapping that we don't want.

This didn't happen in my prior PR for csi-driver-shared-resource: https://github.com/openshift/release/pull/20967

@openshift/test-platform why does `make openshift-image-mirror-mappings` want to add a mirrored image here?

Comment (gabemontero) [reply core-services/image-mirroring/openshift/mapping_origin_4_10]: in the interim @adambkaplan I've updated this PR for 
1) only update master branch ... assuming CI will handle mirroring to 4.10 once that master / 4.10 split occurs
2) changed the name from scripts to e2es since that is what we are dealing with here

let's see how the test jobs here react

**Comment by akram** [comment]
@gabemontero impressive as usual 👍 
@adambkaplan I think that Gabe fixed the mirroring thing now. Can it be LGTMed or I am still missing something?

**Comment by gabemontero** [comment]
> @gabemontero impressive as usual +1
> @adambkaplan I think that Gabe fixed the mirroring thing now. Can it be LGTMed or I am still missing something?

thanks @akram

actually the mirroring stuff is not fixed ... there is this 4.10 only mirror need @adambkaplan and I don't understand given what he did previously for the csi driver stuff

have tried to engage with DPTP to get help on what we are doing wrong here ... will follow up later, but I also want to take another pass and looking at other e2e via step registry usages and the CI doc to see if I can figure out the "correct" path here

File: core-services/image-mirroring/openshift/mapping_origin_4_10
Comment (gabemontero) [reply core-services/image-mirroring/openshift/mapping_origin_4_10]: > Image mirror mapping jobs shouldn't fail here - it seems that the make command logic is trying to create a mirror mapping that we don't want.
> 
> This didn't happen in my prior PR for csi-driver-shared-resource: #20967
> 
> @openshift/test-platform why does `make openshift-image-mirror-mappings` want to add a mirrored image here?

hey @adambkaplan went back and revisited this, and I see from your csi-driver-shared-resource work https://github.com/openshift/release/blob/1f41c71e615c06be295abfd2d6e88cbfa995dd32/core-services/image-mirroring/openshift/mapping_origin_4_10#L89 

`registry.ci.openshift.org/origin/4.10:csi-driver-shared-resource-scripts quay.io/openshift/origin-csi-driver-shared-resource-scripts:4.10 quay.io/openshift/origin-csi-driver-shared-resource-scripts:4.10.0 quay.io/openshift/origin-csi-driver-shared-resource-scripts:latest`

is this not an indication that your changes either *DID* or since your PR merge now *DO* in fact require a mirrored image

and we should move forward with merging this update for e2e tests for jenkins-sync-plugin with a mirrored image change (that has to be approved by Clayton).

**Comment by gabemontero** [comment]
> > @gabemontero impressive as usual +1
> > @adambkaplan I think that Gabe fixed the mirroring thing now. Can it be LGTMed or I am still missing something?
> 
> thanks @akram
> 
> actually the mirroring stuff is not fixed ... there is this 4.10 only mirror need @adambkaplan and I don't understand given what he did previously for the csi driver stuff
> 
> have tried to engage with DPTP to get help on what we are doing wrong here ... will follow up later, but I also want to take another pass and looking at other e2e via step registry usages and the CI doc to see if I can figure out the "correct" path here

See https://github.com/openshift/release/pull/21474#discussion_r719726663

in fact mirroring to quay.io if not originally, appears to now be in place for the recently added csi driver shared resource e2e's 

and hence we should move forward on that path for this one

**Comment by gabemontero** [comment]
ok @adambkaplan @akram all the CI wiring worked fine, but one of the new jenkins e2e's failed because it is now violating the k8s name len constraint:

```
ERROR: create returned an error;
        {err=The Route "nodejs-postgresql-example" is invalid: spec.host: Invalid value: "nodejs-postgresql-example-jenkins-sync-plugin-test-namespace-zxpwk.apps.ci-op-2957rdxc-a7f97.origin-ci-int-aws.dev.rhcloud.com": must be no more than 63 characters
        , verb=create, cmd=oc --server=https://172.30.0.1:443 --certificate-authority=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt --namespace=jenkins-sync-plugin-test-namespace-zxpwk --token=XXXXX create -f /tmp/workspace/jenkins-sync-plugin-test-namespace-zxpwk/jenkins-sync-plugin-test-namespace-zxpwk-sync-plugin-nodejs-declarative-builds/create14323622599870266539.markup -o=name , out=secret/nodejs-postgresql-example
        service/nodejs-postgresql-example
```

In theory this should/could be happening in PRs against openshift/jenkins or running e2e-aws-jenkins in other repos.

We've hit this name too long issue with sample e2es in some form in the past, I just don't remember all the history and what we had to do to solve off the top of my head.  I'll need to do some digging, but most likely I'll need to update jenkins-sync-plugin repo before we'll see the rehearsals fully pass.

**Comment by gabemontero** [comment]
> ok @adambkaplan @akram all the CI wiring worked fine, but one of the new jenkins e2e's failed because it is now violating the k8s name len constraint:
> 
> ```
> ERROR: create returned an error;
>         {err=The Route "nodejs-postgresql-example" is invalid: spec.host: Invalid value: "nodejs-postgresql-example-jenkins-sync-plugin-test-namespace-zxpwk.apps.ci-op-2957rdxc-a7f97.origin-ci-int-aws.dev.rhcloud.com": must be no more than 63 characters
>         , verb=create, cmd=oc --server=https://172.30.0.1:443 --certificate-authority=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt --namespace=jenkins-sync-plugin-test-namespace-zxpwk --token=XXXXX create -f /tmp/workspace/jenkins-sync-plugin-test-namespace-zxpwk/jenkins-sync-plugin-test-namespace-zxpwk-sync-plugin-nodejs-declarative-builds/create14323622599870266539.markup -o=name , out=secret/nodejs-postgresql-example
>         service/nodejs-postgresql-example
> ```
> 
> In theory this should/could be happening in PRs against openshift/jenkins or running e2e-aws-jenkins in other repos.
> 
> We've hit this name too long issue with sample e2es in some form in the past, I just don't remember all the history and what we had to do to solve off the top of my head. I'll need to do some digging, but most likely I'll need to update jenkins-sync-plugin repo before we'll see the rehearsals fully pass.

History found:  https://github.com/openshift/origin/pull/25877

I believe I need to approximate that fix in the jenkins-sync-plugin e2e's @akram @adambkaplan 

Not 100% yet if it will impact e2e-aws-jenkins in openshift/origin or openshift/jenkins ... if there are failures in @adambkaplan 's https://github.com/openshift/jenkins/pull/1324 I'll see if there are any similarity

**Comment by gabemontero** [comment]
https://github.com/openshift/jenkins-sync-plugin/pull/403 once it merges should fix the rehearsal jobs here

**Comment by gabemontero** [comment]
the 4.10 version of the https://github.com/openshift/jenkins-sync-plugin/pull/403 fix wasn't created off of master yet

maybe I need a test PR in that repo to fully flush the pipeline?

let's see

/retest

**Pull request merged**
